src/libstd/io/stdio.rs

```diff
@@ -872,6 +872,95 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Equivalent to the [`input`] method except that
+/// a prompt is printed before capturing input.
+///
+/// ## Note
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
+///
+/// ## Panics
+/// This function currently panics if it does not receive input,
+/// or when input is empty without newline
+///
+/// # Examples
+///
+/// ```
+/// #![feature(input_prompt)]
+/// use std::io::input_prompt;
+///
+/// fn main() {
+///     let user_input = input_prompt("Please enter some text: ");
+///
+///     println!("You typed: {}", user_input);
+/// }
+/// ```
+#[unstable(
+    feature = "input_prompt",
+    reason = "this function may be replaced with a \
+                     more general mechanism",
+    issue = "none"
+)]
+pub fn input_prompt(prompt: &str) -> String {
+    stdout().write(prompt.as_bytes()).unwrap();
+    input()
+}
+
+/// Constructs a new handle to the standard input of the current
+/// process, locks this handle and reads a line of input,
+/// returning a `String` containing the input.
```

Comment:
I think there should be a link to `prompt`.

Reply:
Added link to `input_prompt`, so both methods refer to each other.